### DISCUSSION TITLE
Change field to display in relation select

### DISCUSSION THREAD

**Discussion by Ashot-KR** [strapi/strapi#7543] [Legacy Topics]
Q1: How can i change displayed entity title in relations select? I have changed `entity title` on "Configure view" screen but it have no effect
versions:
```
strapi@3.1.4
strapi-admin@3.1.4
```
![image](https://user-images.githubusercontent.com/899218/90826793-c378a880-e343-11ea-8ea8-5a7023e2bccf.png)

Q2: can displayed titles in this selects be customised?
For example, i have entity `Author`, it have `firstName` and `lastName` fields. I want to display `${firstName} ${lastName}` as option text. Is it possible?

**Comment by derrickmehaffy** [comment]
Your first question, that is odd as it should be working properly, you are in the right place. Is your project code somewhere I can look at, and what database type are you using.

Your second question is there is no option for that built in, it would require you to add some extensions to the `strapi-plugin-content-manager` plugin

**Reply by Ashot-KR** [reply]
> Your first question, that is odd as it should be working properly, you are in the right place. Is your project code somewhere I can look at, and what database type are you using.

database is SQLite. Code is not hosted anywhere. Actually it's not a project, i'm just tasting strapi. 😄 
I'll try to find some free time tomorrow to create example repo.
I've made some investigation:
1. original `mainField` is `firstName` (ex. Brandon)
2. i change it to `lastName` (ex. Sanderson)
3. this value saved properly, also it saved in DB (table `core_store` `key=plugin_content_manager_configuration_content_types::application::book.book`)
4. in relation select `firstName` still displayed (see screenshot above)

some kind of cache maybe?

**Reply by derrickmehaffy** [reply]
That's what it looks like, but I wasn't able to reproduce it locally on a fresh v3.1.4 project

**Reply by Ashot-KR** [reply]
https://github.com/Ashot-KR/strapi-test
start app
login: `wat@wat.com` / `123qweASD`
go to http://localhost:1337/admin/plugins/content-manager/collectionType/application::author.author/ctm-configurations/edit-settings/content-types – entity title value is `lastName`
go to http://localhost:1337/admin/plugins/content-manager/collectionType/application::book.book/1 – there is `brandon` displayed in relations (this is authors `firstName`)

**Reply by derrickmehaffy** [reply]
Thank you, let me take a look :)

**Reply by derrickmehaffy** [reply]
Hi @Ashot-KR 

I wasn't able to reproduce the issue but I did find a workaround for your 2nd question, the documentation I was referencing is here: https://strapi.io/documentation/v3.x/concepts/models.html#lifecycle-hooks

And here is a video of me going through the process: https://www.loom.com/share/f45e106174b043788b92a2d8ceb5dd3c

(Sorry the bad audio, had the wrong mic selected)

**Reply by Ashot-KR** [reply]
i was doing it in wrong place 🤦 😄 
![image](https://user-images.githubusercontent.com/899218/90921030-07c28200-e3f2-11ea-8b6f-2f7209313591.png)

> the documentation I was referencing is here: https://strapi.io/documentation/v3.x/concepts/models.html#lifecycle-hooks

i will take a look

thanks for your help! 🍻

**Reply by derrickmehaffy** [reply]
No problem, good luck!

**Reply by Ashot-KR** [reply]
@derrickmehaffy i found another way to implement Q2 (and i think it's more elegant)

#### Step 1
install `bookshelf-virtuals-plugin` plugin
```
npm i bookshelf-virtuals-plugin
```
add plugin to `bookshelf`
```js
// config/funcitons/bookshelf.js

module.exports = (bookshelf, connection) => {
  bookshelf.plugin('bookshelf-virtuals-plugin');
};
```
#### Step 2
Add virtual field to model
```js
// api/author/models/author.js

module.exports = {
  options: {
    virtuals: {
      fullName () {
        return `${this.get('firstName')} ${this.get('lastName')}`
      }
    }
  }
}
```
#### Step 3
Add `fullName` to model config
```js
// api/author/models/author.settings.json

"attributes" : {
  "fullName": {
    "type": "string",
    "required": false,
    "configurable": false // to disable edit from content-type editor, optional
  },
}
```
Now we able to choose `fullName` field as entity title.

**It works only with SQL databases, not with MongoDB!**
Actually `mongoose` has built-in support of virtual fields, but I can't find way to set virtual fields for mongoose models.
